test/csuite/tiered_abort/main.c

```diff
@@ -0,0 +1,941 @@
+        }
+
+        cur_coll->set_key(cur_coll, kname);
+        cur_local->set_key(cur_local, kname);
+        cur_oplog->set_key(cur_oplog, kname);
+        cur_shadow->set_key(cur_shadow, kname);
+        /*
+         * Put an informative string into the value so that it can be viewed well in a binary dump.
+         */
+        testutil_check(__wt_snprintf(cbuf, sizeof(cbuf),
+          "COLL: thread:%" PRIu32 " ts:%" PRIu64 " key: %" PRIu64, td->info, active_ts, i));
+        testutil_check(__wt_snprintf(lbuf, sizeof(lbuf),
+          "LOCAL: thread:%" PRIu32 " ts:%" PRIu64 " key: %" PRIu64, td->info, active_ts, i));
+        testutil_check(__wt_snprintf(obuf, sizeof(obuf),
+          "OPLOG: thread:%" PRIu32 " ts:%" PRIu64 " key: %" PRIu64, td->info, active_ts, i));
+        data.size = __wt_random(&rnd) % MAX_VAL;
+        data.data = cbuf;
+        cur_coll->set_value(cur_coll, &data);
+        testutil_check(pthread_rwlock_rdlock(&flush_lock));
+        locked = true;
+        if ((ret = cur_coll->insert(cur_coll)) == WT_ROLLBACK)
+            goto rollback;
+        testutil_check(ret);
+        cur_shadow->set_value(cur_shadow, &data);
+        if (use_ts) {
+            /*
+             * Change the timestamp in the middle of the transaction so that we simulate a
+             * secondary.
+             */
+            ++active_ts;
+            testutil_check(
+              __wt_snprintf(tscfg, sizeof(tscfg), "commit_timestamp=%" PRIx64, active_ts));
+            testutil_check(session->timestamp_transaction(session, tscfg));
+        }
+        if ((ret = cur_shadow->insert(cur_shadow)) == WT_ROLLBACK)
+            goto rollback;
+        data.size = __wt_random(&rnd) % MAX_VAL;
+        data.data = obuf;
+        cur_oplog->set_value(cur_oplog, &data);
+        if ((ret = cur_oplog->insert(cur_oplog)) == WT_ROLLBACK)
+            goto rollback;
+        testutil_check(session->commit_transaction(session, NULL));
+        /*
+         * Insert into the local table outside the timestamp txn. This must occur after the
+         * timestamp transaction, not before, because of the possibility of rollback in the
+         * transaction. The local table must stay in sync with the other tables.
+         */
+        data.size = __wt_random(&rnd) % MAX_VAL;
+        data.data = lbuf;
+        cur_local->set_value(cur_local, &data);
+        testutil_check(cur_local->insert(cur_local));
+        testutil_check(pthread_rwlock_unlock(&flush_lock));
+        locked = false;
+
+        /* Save the timestamps and key separately for checking later. */
+        if (fprintf(fp, "%" PRIu64 " %" PRIu64 " %" PRIu64 "\n", active_ts,
+              durable_ahead_commit ? active_ts + 4 : active_ts, i) < 0)
+            testutil_die(EIO, "fprintf");
+
+        if (0) {
+rollback:
+            testutil_check(session->rollback_transaction(session, NULL));
+            if (locked) {
+                testutil_check(pthread_rwlock_unlock(&ts_lock));
```

Comment:
It looks like this should unlock `flush_lock`?

Reply:
Good catch! Yes it should. Fixed.